File: Leetcode/560 - Subarray Sum Equals K/solution.js
var subarraySum = function(nums, k) {
    const prefixSum = new Map();
    prefixSum.set(0, 1);

    let sum = 0;
    let count = 0;

    for (const num of nums) {
        sum += num;
        if (prefixSum.has(sum - k)) {
            count += prefixSum.get(sum - k);
        }
        prefixSum.set(sum, (prefixSum.get(sum) || 0) + 1);
    }

    return count;
};

File: Leetcode/560 - Subarray Sum Equals K/solution.py
def subarraySum(self, nums: List[int], k: int) -> int:
    prefix_sum: Dict[int, int] = {0: 1}
    sum_: int = 0
    count: int = 0

    for num in nums:
        sum_ += num
        if (sum_ - k) in prefix_sum:
            count += prefix_sum[sum_ - k]
        prefix_sum[sum_] = prefix_sum.get(sum_, 0) + 1

    return count